docs/developers/HowTo.md

```diff
@@ -0,0 +1,84 @@
+to Gluten as soon as possible, we collected these frequently asked questions, and organized them in the form of Q&A. It's convenient for the developers
+to check and learn.
+
+when you encountered a new problem and then resolved it, please add a new item to this document if you think it may be helpful to the other developers.
+
+# How to understand the key work of Gluten?
+
+The Gluten worked as the role of bridge, it's a middle layer between the Spark and the native execution library.
+
+The Gluten is responsibility for validating whether the operators of the Spark plan can be executed by the native engine or not. If yes, the Gluten
+transforms Spark plan to Substrait plan, and then send the Substrait plan to the native engine.
+
+The Gluten codes consist of two parts: the C++ codes and the Java/Scala codes. 
+1. All C++ codes place under the directory of `gluten_home/cpp`, the Java/Scala codes place under several directories, such as `gluten_home/gluten-core`,
+  `gluten_home/gluten-data`, `gluten_home/backends_velox`.
+2. The Java/Scala codes are responsibility for validating and transforming execution plan. Source data should also be provided, the source data may come
+   from files or other forms such as networks.
+3. The C++ codes take the Substrait plan and the source data as inputs and transform the Substrait plan to the corresponding backend plan. If the backend
+   is Velox, the Substrait plan will be transformed to the Velox plan, and then be executed.
+
+JNI is a programming technology of invoding C++ from Java. All JNI interfaces are defined in the file `JniWrapper.cc` under directory `jni`.
+
+# How to debug in Gluten?
+
+## 1 How to debug C++
+If you don't concern about the Scala/Java codes and just want to debug the C++ codes, you may debug the Velox bench-mark following the below steps.
+
+1. First step, compile cpp with `BUILD_TYPE` debug.
+  - `cd gluten_home/cpp`
+  - `./compile.sh --BUILD_TYPE=debug`. 
+  compiling with `BUILD_TYPE=debug` is good for debugging, after the above two operations, the executable file `generic_benchmark` will be generated under
+  the directory of `gluten_home/cpp/velox/benchmarks/`.
+
+2. Second step, debug benchmarks using GDB
+  - `cd gluten_home/cpp/velox/benchmarks/`
+  Now, debug benchmarks with GDB, execute the below command:
+  - `gdb generic_benchmark`
+  When GDB load `generic_benchmark` successfully, you can set `breakpoint` on the `main` function with the command `b main`, and then run with the `r` 
+  command of gdb.
+  You can debug `generic_benchmark` using any gdb commands as debugging normal C++ program, because the `generic_benchmark` is a pure C++ executable file.
+
+3. `gdb-tui` is a valuable feature and is worth trying. You can get more help from online docs.
+  [gdb-tui](https://sourceware.org/gdb/current/onlinedocs/gdb/TUI.html#TUI)
+
+4. You can start `generic_benchmark` with specific JSON plan and input files.
```

Comment:
It's better to clarify that user need to build gluten java first, because the example files won't be genrated if we only build gluten cpp.

Reply:
OK， thanks for your advice